### PR TITLE
You can now add the option writeStream to the reporters.

### LINKED ISSUE #24: Add option writeStream to reporters
See discussion in https://github.com/stefanjudis/grunt-jsinspect/pull/29

Starting point would be here: https://github.com/danielstjules/jsinspect/blob/master/lib/reporters/base.js#L16

### PR DESCRIPTION
**Pull request by borisdiakur**
var reporter = new JSONReporter(inspector, {
      writeStream: customWriteStream
    });

This PR resolves #24

### CODE REVIEW

**Comment by danielstjules** [comment]
Thanks a lot for your work on this! :) Looks really good. Just a coupe small things:
* The `self._writeStream.end();` logic could be moved to the base reporter, since it's common to all
* If writing to a temp file, it'd probably be better to use something like https://www.npmjs.com/package/tmp But we could also avoid writing to disk for the spec. It'd be awesome if it could just something like https://github.com/maxogden/concat-stream

File: package.json [hunk 1]
```diff
@@ -29,13 +29,14 @@
...
+    "concat-stream": "^1.5.0",
+    "expect.js": "*",
+    "mocha": "*"
   },
   "bin": {
     "jsinspect": "./bin/jsinspect"
   },
   "scripts": {
-    "test": "./node_modules/.bin/mocha -R spec spec spec/reporters"
+    "test": "./node_modules/.bin/mocha -R spec spec/reporters"
```
Comment (danielstjules) [inline package.json]: Can you undo this? This change makes it so that none of other unit tests (nodeUtils, Match, Inspector) run

**Comment by danielstjules** [comment]
Thanks again. If you squash I'll merge it in :)

File: package.json (hunk 1 above)
Comment (borisdiakur) [inline package.json]: :anguished:  Sorry about that one. I haven't checked carefully enough.

**Comment by borisdiakur** [comment]
squash :heavy_check_mark:

**Comment by danielstjules** [comment]
Looks like you still have 3 commits here? Did you forget to force push to this branch after [squashing](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)?

**Comment by borisdiakur** [comment]
squash :heavy_check_mark:

**Comment by danielstjules** [comment]
Thanks again! :)

**Comment by danielstjules** [comment]
Tagged a new release: https://github.com/danielstjules/jsinspect/releases/tag/0.7.0 :tada: 

As a note, I renamed the option to `opts.writableStream`, to be consistent with [the class name](https://nodejs.org/api/stream.html#stream_class_stream_writable)

**Pull request merged**